Tests/MapboxMapsTests/Foundation/Mocks/MockStyleManager.swift

```diff
@@ -0,0 +1,462 @@
+        defaultReturnValue: .init(value: "stub", kind: .undefined)
+    )
+    func getStyleLightProperty(forProperty property: String) -> MapboxCoreMaps.StylePropertyValue {
+        getStyleLightPropertyStub.call(with: property)
+    }
+
+    let getStyleTerrainPropertyStub = Stub<String, MapboxCoreMaps.StylePropertyValue>(
+        defaultReturnValue: .init(value: "stub", kind: .undefined)
+    )
+    func getStyleTerrainProperty(forProperty property: String) -> MapboxCoreMaps.StylePropertyValue {
+        getStyleTerrainPropertyStub.call(with: property)
+    }
+
+    let getStyleProjectionPropertyStub = Stub<String, MapboxCoreMaps.StylePropertyValue>(
+        defaultReturnValue: .init(value: "stub", kind: .undefined)
+    )
+    func getStyleProjectionProperty(forProperty property: String) -> MapboxCoreMaps.StylePropertyValue {
+        getStyleTerrainPropertyStub.call(with: property)
+    }
+
+    let getStyleImageStub = Stub<String, MapboxCoreMaps.Image?>(defaultReturnValue: nil)
+    func getStyleImage(forImageId imageId: String) -> MapboxCoreMaps.Image? {
+        getStyleImageStub.call(with: imageId)
+    }
+
+    let hasStyleImageStub = Stub<String, Bool>(defaultReturnValue: false)
+    func hasStyleImage(forImageId imageId: String) -> Bool {
+        hasStyleImageStub.call(with: imageId)
+    }
+
+    let isStyleLoadedStub = Stub<Void, Bool>(defaultReturnValue: false)
+    func isStyleLoaded() -> Bool {
+        isStyleLoadedStub.call()
+    }
+
+    // MARK: Layers
+
+    struct AddStyleLayerParameters {
+        let properties: Any
+        let layerPosition: MapboxCoreMaps.LayerPosition?
+    }
+    let addStyleLayerStub = Stub<AddStyleLayerParameters, Expected<NSNull, NSString>>(
+        defaultReturnValue: .init(value: NSNull())
+    )
+    func addStyleLayer(
+        forProperties properties: Any,
+        layerPosition: MapboxCoreMaps.LayerPosition?
+    ) -> Expected<NSNull, NSString> {
+
+        addStyleLayerStub.call(with: AddStyleLayerParameters(properties: properties, layerPosition: layerPosition))
+    }
+
+    struct AddStyleCustomLayerParameters {
+        let layerId: String
+        let layerHost: CustomLayerHost
+        let layerPosition: MapboxCoreMaps.LayerPosition?
+    }
+    let addStyleCustomLayerStub = Stub<AddStyleCustomLayerParameters, Expected<NSNull, NSString>>(
+        defaultReturnValue: .init(value: NSNull())
+    )
+    func addStyleCustomLayer(
+        forLayerId layerId: String,
+        layerHost: CustomLayerHost,
+        layerPosition: MapboxCoreMaps.LayerPosition?
+    ) -> Expected<NSNull, NSString> {
+
+        addStyleCustomLayerStub.call(
+            with: AddStyleCustomLayerParameters(layerId: layerId, layerHost: layerHost, layerPosition: layerPosition)
+        )
+    }
+
+    let addPersistentStyleLayerStub = Stub<AddStyleLayerParameters, Expected<NSNull, NSString>>(
+        defaultReturnValue: .init(value: NSNull())
+    )
+    @objc func addPersistentStyleLayer(
```

Comment:
is the `@objc` attribute necessary?

Reply:
good catch :-D it is not needed, I added it to try on some swizzle and forgot to remove it